lib/rules/spinner-needs-labelling.js

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+"use strict";
+
+const { hasNonEmptyProp } = require("../util/hasNonEmptyProp");
+const elementType = require("jsx-ast-utils").elementType;
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        // possible error messages for the rule
+        messages: {
+            noUnlabelledSpinner: "Accessibility: Spinner must have aria-label and aria-live"
+        },
+        // "problem" means the rule is identifying code that either will cause an error or may cause a confusing behavior: https://eslint.org/docs/latest/developer-guide/working-with-rules
+        type: "problem",
+        // docs for the rule
+        docs: {
+            description: "Accessibility: Spinner must have aria-label and aria-live",
+            recommended: true,
+            url: "https://www.w3.org/TR/html-aria/" // URL to the documentation page for this rule
+        },
+        schema: []
+    },
+    // create (function) returns an object with methods that ESLint calls to “visit” nodes while traversing the abstract syntax tree
+    create(context) {
+        return {
+            // visitor functions for different types of nodes
+            JSXOpeningElement(node) {
+                // if it is not a Spinner, return
+                if (elementType(node) !== "Spinner") {
+                    return;
+                }
+
+                if (hasNonEmptyProp(node.attributes, "aria-label") && hasNonEmptyProp(node.attributes, "aria-live")) {
```

Comment:
I believe that if the loading indicator has text content then we don't need the aria label. Maybe the hasTextContentChild function could be used.

Reply:
fluent implementation has both text content and aria-label for screen reader: https://github.com/microsoft/fluentui/blob/master/packages/react/src/components/Spinner/Spinner.base.tsx.

What do you suggest doing?

Reply:
It actually uses the label prop <Spinner appearance="inverted" label="Inverted Spinner" />. so it's either the label or the aria-label

Reply:
Got it, will do